Scripts/General/ModMercs.lua

```diff
@@ -0,0 +1,732 @@
+        }
+        
+        table.insert(MercSaveData, NewUpgrade)
+
+        Upgrade = table.copy(NewUpgrade)
+    else
+        if Upgrade.Level < const.Mercenary.UpgradeLimit then
+            Upgrade.Level = Upgrade.Level + 1
+        end
+    end
+
+    -- Exception for Summmons: Reset on upgrade
+    if MercUpgradeType == EMercUpgradeType.Charges then
+        MercSaveData.FightsLeft = Merc.FightsMax[Upgrade.Level + 1]
+    end
+
+    return Upgrade
+end
+
+function Merc_Fight(Merc, t)
+
+    evt.Subtract("NPCs", QuestNPC)
+    Merc_ConsumeCharge(Merc)
+
+    local MercSaveData = Merc_GetSaveDataByID(Merc.NPC_ID)
+    
+    MercSaveData.Released       = true
+    MercSaveData.ReleaseMap     = Game.Map.Name
+    
+    local UpgradeMonLevel       = Merc_GetUpgradeLevel(Merc, EMercUpgradeType.Level) + 1
+    local UpgradeHPLevel        = Merc_GetUpgradeLevel(Merc, EMercUpgradeType.HP) + 1
+    local UpgradeACLevel        = Merc_GetUpgradeLevel(Merc, EMercUpgradeType.AC) + 1
+    local UpgradeDmg1Level      = Merc_GetUpgradeLevel(Merc, EMercUpgradeType.MeleeDamage) + 1
+    --local UpgradeDmg1Level      = GetUpgradeLevel(Merc, EMercUpgradeType.MageRes)
+
+    -- Monster
+    local mon                   = SummonMonster(Merc.MonsterID, Party.X, Party.Y, Party.Z, false)
+    
+    -- User values
+    mon.FullHitPoints           = Merc.FullHP[UpgradeHPLevel]
+    mon.ArmorClass              = Merc.AC[UpgradeHPLevel]
+    mon.Level                   = Merc.Level[UpgradeHPLevel]
+
+    -- Default Variables
+    mon.NPC_ID                  = QuestNPC
+    mon.Group                   = 35
+    mon.Hostile                 = false
+    mon.Ally                    = 9999
+    mon.Summoner                = 28
+    mon.HostileType             = 0
+    mon.Experience              = 0
+    mon.ShowOnMap               = true
+    mon.HP                      = mon.FullHitPoints
+    mon.NoFlee                  = true
+    
+    -- Damage
+    local meleeDmg              = Merc_ParseDamageString(Merc.Attack1[UpgradeDmg1Level])
+    mon.Attack1.DamageAdd       = meleeDmg.DamageAdd
+    mon.Attack1.DamageDiceCount = meleeDmg.DamageDiceCount
+    mon.Attack1.DamageDiceSides = meleeDmg.DamageDiceSides
+
+    ExitScreen()
+end
+
+function Merc_Dismiss(Merc, t)
+
+    evt.Add("NPCs", QuestNPC)
+
+    local MercSaveData      = Merc_GetSaveDataByID(Merc.NPC_ID)
+    MercSaveData.Released   = false
+    MercSaveData.ReleaseMap = ""
+    
+    for _, mon in Map.Monsters do
+        if mon.NPC_ID == QuestNPC then
+            RemoveMonster(mon)
+        end
+    end
+
+    -- Reset
+    Timer(
+        function()
+            local UpgradeUseLevel   = Merc_GetUpgradeLevel(Merc, EMercUpgradeType.Charges) or 1
+            local MercSaveData      = Merc_GetSaveDataByID(Merc.NPC_ID)
+            MercSaveData.FightsLeft = Merc.FightsMax[UpgradeUseLevel+1] or 1
+        end, 
+        const.Day, const.Hour, true)
+
+    ExitScreen()
+end
+
+function Merc_IsSpecial(Merc, t)
```

Comment:
So there is a subtype of Merc? Writing a wrapper for such logic is a reasonable thing to do 👍  
But I wonder if this should be documented. I have no idea what "special" may mean.